packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -944,6 +975,9 @@ class _DatePickerDefaultsM2 extends DatePickerThemeData {
         elevation: 24.0,
         shape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(4.0))),
         dayShape: const MaterialStatePropertyAll<OutlinedBorder>(CircleBorder()),
+        yearShape: const MaterialStatePropertyAll<OutlinedBorder>(
+          RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(28.0))),
```

Comment:
I suspect that this probably changes the default shape. The default shape is always a pill shape, which has borderRadius equal to half the height. Using a fixed border radius here should result in a different shape if `textScaleFactor` is customized. 

I think `StadiumBorder` should work.

Reply:
Correct, previously borderRadius was equal to half the height. So, I will use `StadiumBorder` instead.